domain/bankaccount.py

```diff
@@ -1,43 +1,68 @@
 from typing import List
-from domain.transaction import Transaction, create_transaction
-from domain.commontypes.types import AccountID
 
-Amount = float
+from domain.ledgeraccount import LedgerAccount
+from domain.transaction import Transaction, create_transaction
+from domain.commontypes.types import AccountID, Amount
 
 
 class BankAccount:
     def __init__(self,
                  account_id: AccountID,
-                 transactions: List[Transaction]
+                 main_account: LedgerAccount,
+                 draft_account: LedgerAccount,
+                 stage_account: LedgerAccount,
+                 overdraft_limit: Amount
                  ):
         self._id = account_id
-        self._transactions = transactions
+        self._main_account = main_account
+        self._draft_account = draft_account
+        self._stage_account = stage_account
+        self._overdraft_limit = overdraft_limit
 
     def get_balance(self) -> Amount:
-        balance = 0.0
-        for t in self._transactions:
-            if t.d_acc == self._id:
-                balance -= t.value
-            else:
-                balance += t.value
-        return balance
+        main_balance = self._main_account.get_balance()
+        draft_balance = self._draft_account.get_balance()
+        return main_balance - draft_balance
 
     def transfer(self, destiny_id: AccountID, value: Amount) -> None:
         if value <= 0:
```

Comment:
Use the same type (Amount) to a possible negative quantity (the return of get_balance on a bank Account) and a non-negative quantity (value argument in the transfer method) forces that kind of verification.

Reply:
I did not understand... Should I use a Maybe[Amount] or something like that?

Reply:
No. Use two types, one to represent Positive only values, and use them on transfer argument, them you can remove this if. On possible-negative values, like bankAccount balance, use a different type.